llama-index-core/llama_index/core/chat_engine/flow.py

```diff
@@ -0,0 +1,320 @@
+        text_qa_template: Optional[str] = None,
+        llm: Optional[LLM] = None,
+        **kwargs: Any,
+    ) -> "FlowChatEngine":
+        """Initialize a FlowChatEngine from default parameters."""
+        llm = llm or llm_from_settings_or_context(Settings, service_context)
+
+        chat_history = chat_history or []
+        memory = memory or ChatMemoryBuffer.from_defaults(
+            chat_history=chat_history, token_limit=llm.metadata.context_window - 256
+        )
+
+        if system_prompt is not None:
+            if prefix_messages is not None:
+                raise ValueError(
+                    "Cannot specify both system_prompt and prefix_messages"
+                )
+            prefix_messages = [
+                ChatMessage(content=system_prompt, role=llm.metadata.system_role)
+            ]
+
+        prefix_messages = prefix_messages or []
+        node_postprocessors = node_postprocessors or []
+
+        return cls(
+            retriever,
+            llm=llm,
+            memory=memory,
+            prefix_messages=prefix_messages,
+            node_postprocessors=node_postprocessors,
+            callback_manager=callback_manager_from_settings_or_context(
+                Settings, service_context
+            ),
+            text_qa_template=text_qa_template,
+        )
+
+    def _generate_qa_context(self, message: str) -> Tuple[str, List[NodeWithScore]]:
+        """Generate context information from a message."""
+        nodes = self._retriever.retrieve(message)
+        for postprocessor in self._node_postprocessors:
+            nodes = postprocessor.postprocess_nodes(
+                nodes, query_bundle=QueryBundle(message)
+            )
+
+        context_str = "\n\n".join(
+            [n.node.get_content(metadata_mode=MetadataMode.LLM).strip() for n in nodes]
+        )
+
+        return (
+            self._text_qa_template.format(context_str=context_str, query_str=message),
+            nodes,
+        )
+
+    async def _agenerate_qa_context(
+        self, message: str
+    ) -> Tuple[str, List[NodeWithScore]]:
+        """Generate context information from a message."""
+        nodes = await self._retriever.aretrieve(message)
+        for postprocessor in self._node_postprocessors:
+            nodes = postprocessor.postprocess_nodes(
+                nodes, query_bundle=QueryBundle(message)
+            )
+        context_str = "\n\n".join(
+            [n.node.get_content(metadata_mode=MetadataMode.LLM).strip() for n in nodes]
+        )
+
+        return (
+            self._context_template.format(context_str=context_str, query_str=message),
+            nodes,
+        )
+
+    @trace_method("chat")
+    def chat(
+        self, message: str, chat_history: Optional[List[ChatMessage]] = None
+    ) -> AgentChatResponse:
+        if chat_history is not None:
+            self._memory.set(chat_history)
+
+        qa_context_str_template, nodes = self._generate_qa_context(message)
```

Comment:
Do you see any issues with only using the most recent message to retrieve? (tbh, this was kind of the key reason for introducing the condense_plus_context chat engine. Maybe we can bundle similar logic here under a boolean and avoid yet another class for this mode?)